once_test.py

```diff
@@ -792,21 +789,36 @@ async def async_yielding_iterator():
         self.assertEqual([i async for i in async_yielding_iterator()], [1, 2, 3])
         self.assertEqual([i async for i in async_yielding_iterator()], [1, 2, 3])
 
-    @unittest.skip("This currently hangs and needs to be fixed, GitHub Issue #12")
     async def test_failing_generator(self):
         counter = Counter()
 
         @once.once
         async def sample_failing_fn():
             yield counter.get_incremented()
-            raise ValueError("expected failure")
+            result = counter.get_incremented()
+            yield result
+            if result == 2:
+                raise ValueError("expected failure after 2.")
 
+        # Both of these calls should return the same results.
+        call1 = sample_failing_fn()
+        call2 = sample_failing_fn()
+        self.assertEqual(await anext(call1), 1)
+        self.assertEqual(await anext(call2), 1)
+        self.assertEqual(await anext(call1), 2)
+        self.assertEqual(await anext(call2), 2)
         with self.assertRaises(ValueError):
-            [i async for i in sample_failing_fn()]
+            await anext(call1)
         with self.assertRaises(ValueError):
-            [i async for i in sample_failing_fn()]
-        self.assertEqual(await anext(sample_failing_fn()), 1)
-        self.assertEqual(await anext(sample_failing_fn()), 1)
+            await anext(call2)
+        # These next 2 calls should succeed.
+        call3 = sample_failing_fn()
+        call4 = sample_failing_fn()
+        self.assertEqual([i async for i in call3], [3, 4])
+        self.assertEqual([i async for i in call4], [3, 4])
```

Comment:
Based on our API conversations, these calls should return 1, 2, ValueError

Reply:
made configurable!